var/spack/repos/builtin/packages/icu4c/package.py

```diff
@@ -104,7 +101,12 @@ def configure_args(self):
         return args
 
 
-class MSBuildBuilder(spack.build_systems.msbuild.MSBuildBuilder, BuildEnvironment):
+class MSBuildBuilder(spack.build_systems.msbuild.MSBuildBuilder):
+    # Need to make sure that locale is UTF-8 in order to process source files in UTF-8.
+    @when("@59:")
```

Comment:
Is the idiom now to duplicate callbacks inside each builder?
(also this was my PR, sorry if it broke things)

Reply:
Let's say `@when` focuses too much on an edge case that complicates the common use case... 

It only works combined with `MultiMethodMeta`. For example

```python
class foo(metaclass=MultiMethodMeta):
  @when("condition 1")
  def bar(self):
    return 1

  @when("condition 2")
  def bar(self):
    return 2
```

lets `foo.bar` conditionally dispatch the either of the `bar` implementations. In normal class definitions `foo.bar` just binds to the latter one, the former is redundant/dropped.

Even if you don't make use of this, you apparently need MultiMethodMeta.